source/common/network/connection_impl.cc

```diff
@@ -474,6 +486,11 @@ void ConnectionImpl::updateWriteBufferStats(uint64_t num_written, uint64_t new_s
                                            connection_stats_->write_current_);
 }
 
+bool ConnectionImpl::isDoubleHalfClosed() {
```

Comment:
Would bothSidesClosed be more clear?  Somehow I was reading DoubleHalfClosed as sending a FIN  2x on one path but maybe that's just me :-)

Reply:
Yes, that sounds like a better name.